File: app/src/main/java/com/example/deltatask3/api/PokemonApi.java
package com.example.deltatask3.api;

import com.example.deltatask3.utils.EvolutionChain;
import com.example.deltatask3.utils.ItemLocation;
import com.example.deltatask3.utils.Pokedex;
import com.example.deltatask3.utils.Pokemon;
import com.example.deltatask3.utils.PokemonID;
import com.example.deltatask3.utils.PokemonSpecies;
import com.example.deltatask3.utils.Region;
import com.example.deltatask3.utils.SearchResult;
import com.example.deltatask3.utils.Type;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

public interface PokemonApi {

    @GET("https://pokeapi.co/api/v2/pokemon/")
    Call<SearchResult> getPokemonWithOffsetAndLimit(@Query("offset") int offset, @Query("limit") int limit);

    @GET("https://pokeapi.co/api/v2/pokemon/{name}/")
    Call<Pokemon> getPokemonFromName(@Path("name") String name);

    @GET("https://pokeapi.co/api/v2/pokemon-species/{id}/")
    Call<PokemonSpecies> getSpeciesFromID(@Path("id") int id);

    @GET("https://pokeapi.co/api/v2/evolution-chain/{id}")
    Call<EvolutionChain> getEvolutionChainFromID(@Path("id") String idPlusSlash);

    @GET("https://pokeapi.co/api/v2/pokemon/{name}/")
    Call<PokemonID> getPokemonIDFromName(@Path("name") String name);

    @GET("https://pokeapi.co/api/v2/region/{id}/")
    Call<Region> getRegionFromID(@Path("id") int id);

    @GET("https://pokeapi.co/api/v2/pokedex/{name}/")
    Call<Pokedex> getPokedexFromName(@Path("name") String name);

    @GET("https://pokeapi.co/api/v2/type/{id}/")
    Call<Type> getTypeFromID(@Path("id") int id);

    @GET("https://pokeapi.co/api/v2/item/")
    Call<SearchResult> getItemsWithOffsetAndLimit(@Query("offset") int offset, @Query("limit") int limit);

    @GET("https://pokeapi.co/api/v2/item/{name}/")
    Call<ItemLocation> getItemFromName(@Path("name") String name);

    @GET("https://pokeapi.co/api/v2/location/")
    Call<SearchResult> getLocationsWithOffsetAndLimit(@Query("offset") int offset, @Query("limit") int limit);

    @GET("https://pokeapi.co/api/v2/location/{name}/")
    Call<ItemLocation> getLocationFromName(@Path("name") String name);

    @GET("https://pokeapi.co/api/v2/pokemon/{id}/")
    Call<Pokemon> getPokemonFromID(@Path("id") int id);

    @GET("https://pokeapi.co/api/v2/item/{id}/")
    Call<ItemLocation> getItemFromID(@Path("id") int id);

    @GET("https://pokeapi.co/api/v2/location/{id}/")
    Call<ItemLocation> getLocationFromID(@Path("id") int id);
}

File: app/src/main/java/com/example/deltatask3/api/PokemonApi.kt
package com.example.deltatask3.api

import com.example.deltatask3.models.*
import retrofit2.Call
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

interface PokemonApi {
    @GET("pokemon")
    suspend fun getPokemon(@Query("offset") offset: Int, @Query("limit") limit: Int): Response<SearchResult>

    @GET("pokemon/{name}/")
    suspend fun getPokemon(@Path("name") name: String): Response<Pokemon>

    @GET("pokemon/{id}/")
    suspend fun getPokemon(@Path("id") id: Int): Response<Pokemon>

    @GET("pokemon-species/{id}/")
    fun getSpecies(@Path("id") id: Int): Call<PokemonSpecies>

    @GET("evolution-chain/{id}")
    fun getEvolutionChain(@Path("id") idPlusSlash: String): Call<EvolutionChain>

    @GET("pokemon/{name}/")
    fun getPokemonId(@Path("name") name: String): Call<PokemonId>

    @GET("region/{id}/")
    suspend fun getRegion(@Path("id") id: Int): Response<Region>

    @GET("pokedex/{name}/")
    suspend fun getPokedex(@Path("name") name: String): Response<Pokedex>

    @GET("type/{id}/")
    suspend fun getType(@Path("id") id: Int): Response<Type>

    @GET("item")
    suspend fun getItems(@Query("offset") offset: Int, @Query("limit") limit: Int): Response<SearchResult>

    @GET("item/{name}/")
    suspend fun getItem(@Path("name") name: String): Response<ItemLocation>

    @GET("item/{id}/")
    suspend fun getItem(@Path("id") id: Int): Response<ItemLocation>

    @GET("location")
    suspend fun getLocations(@Query("offset") offset: Int, @Query("limit") limit: Int): Response<SearchResult>

    @GET("location/{name}/")
    suspend fun getLocation(@Path("name") name: String): Response<ItemLocation>

    @GET("location/{id}/")
    suspend fun getLocation(@Path("id") id: Int): Response<ItemLocation>
}